src/Facility.Core/Http/HttpClientServiceSettings.cs

```diff
@@ -30,6 +30,13 @@ public sealed class HttpClientServiceSettings
 	/// </summary>
 	public HttpContentSerializer? TextSerializer { get; set; }
 
+	/// <summary>
+	/// An optional callback function that determines if a request should be compressed.
+	/// </summary>
+	/// <remarks>Request bodies will be compressed with <c>Content-Encoding: gzip</c>. Even when this callback
+	/// returns <c>true</c>, the request may be sent uncompressed if compressing would make it larger.</remarks>
+	public Func<ServiceDto, bool>? ShouldCompressRequest { get; set; }
```

Comment:
I'm open to just going with `public bool? CompressRequests { get; set; }` if the delegate is YAGNI.

Reply:
I can't think of an _actual_ reason to enable compression on a per-request basis, so happy to switch to the simpler API.

Reply:
Great! We can always add this if we need it.

Reply:
> I can't think of an actual reason to enable compression on a per-request basis

I thought of one: the API might know that `POST /v1/smallData` always has a very small payload (and shouldn't be compressed), but that `POST /v1/bigData` is large and always should be. This could still be YAGNI, though.